spec/lib/guardian_extension_spec.rb

```diff
@@ -77,4 +77,40 @@
       end
     end
   end
+
+  describe "#can_detect_language?" do
+    fab!(:group)
+    fab!(:user) { Fabricate(:user, groups: [group]) }
+    fab!(:post) { Fabricate(:post, user: user, raw: "Hello, world!") }
+    let(:guardian) { Guardian.new(user) }
+
+    it "returns false when the post user is not in restrict_translation_by_poster_group" do
+      SiteSetting.restrict_translation_by_poster_group = "#{group.id + 1}"
```

Comment:
Making artificial IDs can end up making flaky tests, we should either `Fabricate(:group)` here to get a new group or do `group.remove(user)` in this test

Reply:
Yes it matters, I just know pretty much every time anyone's used an ID that wasn't from a DB model in specs, it has led to flakes especially in parallel testing environments. It's just safer to Fabricate a new one, as you've done now :)